### PR TITLE
Update direct download button state on beatmapset removal

### LINKED ISSUE #4232: osu!direct doesn't update immediately
**Describe your problem:**
osu!direct doesn't update the beatmap states immediately.
For example, osu!direct still shows a beatmap as downloaded even though it's already deleted.
This is easily fixed by restarting the game, but it would be nice if it updated immediately.

**Screenshots or videos showing encountered issue:** 
dotnet_2019-02-09_19-24-43
dotnet_2019-02-09_19-25-15
Clicking on the green button will show the beatmap selection screen (and prompt the dialog since there are no beatmaps)

**osu!lazer version:** https://github.com/ppy/osu/commit/7ec0b4ba773180ad8d9c73004dbf1d766df6d38e

**Logs:** (irrelevant)

### PR DESCRIPTION
**Pull request by UselessToucan**
Closes #4232 
https://youtu.be/eHTT4u4SPyA

### CODE REVIEW

File: osu.Game/Overlays/Direct/DownloadButton.cs [hunk 1]
```diff
@@ -76,6 +76,9 @@ private void load(OsuColour colours, OsuGame game, BeatmapManager beatmaps)
         {
             this.colours = colours;
 
+            beatmaps.ItemAdded += (set, existing, silent) => updateState(BeatmapSet.Value);
```
Comment (peppy) [inline osu.Game/Overlays/Direct/DownloadButton.cs]: why is this added? this logic is already in the underlying class, no?

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Direct/DownloadButton.cs (hunk 1 above)
Comment (UselessToucan) [reply osu.Game/Overlays/Direct/DownloadButton.cs]: Sorry, I overlooked that :(

File: osu.Game/Overlays/Direct/DownloadTrackingComposite.cs [hunk 2]
```diff
@@ -120,12 +121,16 @@ private void onRequestFailure(Exception e)
             Schedule(() => attachDownload(null));
         }
 
-        private void setAdded(BeatmapSetInfo s, bool existing, bool silent)
+        private void setAdded(BeatmapSetInfo s, bool existing, bool silent) => setDownloadState(s, DownloadState.Downloaded);
```
Comment (smoogipoo) [inline osu.Game/Overlays/Direct/DownloadTrackingComposite.cs]: Shouldn't this be `LocallyAvailable`?

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Overlays/Direct/DownloadTrackingComposite.cs (hunk 2 above)
Comment (UselessToucan) [reply osu.Game/Overlays/Direct/DownloadTrackingComposite.cs]: yes, it should. will fix this ~10 hours later.

**Comment by peppy** [comment]
Already fixed, along with one other issue (conditional check was not inside schedule call).

**Review by peppy — APPROVED** [review]
